src/modules/job-manager/plugins/limit-duration.c

```diff
@@ -0,0 +1,323 @@
+/************************************************************\
+ * Copyright 2022 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+/* limit-duration.c - validate job requests against configured duration limits
+ *
+ * General limit:
+ *  [policy.limits.duration]
+ * Queue-specific limit:
+ *  [queues.<name>.policy.limits.duration]
+ *
+ * N.B. a queue limit may override the general limit with a higher or lower
+ * limit, even "unlimited" (0).
+ *
+ * See also:
+ *  RFC 33/Flux Job Queues
+ */
```

Comment:
would it be worth mentioning `policy.jobspec.defaults.system.duration`, just to say?